packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeRate.ts

```diff
@@ -114,6 +128,52 @@ export const getTradeRate = async (
     return { networkFeeCryptoBaseUnit: fast.txFee }
   }
 
+  const buyAssetAddress =
+    buyAsset.assetId === solAssetId
+      ? fromAssetId(wrappedSolAssetId).assetReference
+      : fromAssetId(buyAsset.assetId).assetReference
+
+  const sellAssetAddress =
+    sellAsset.assetId === solAssetId
+      ? fromAssetId(wrappedSolAssetId).assetReference
+      : fromAssetId(sellAsset.assetId).assetReference
+
+  const [buyAssetAccount] = await PublicKey.findProgramAddressSync(
```

Comment:
suggestion: should we use `programAddress` terminology? Without being familiar with Solana specifics or without reading this through carefully, one could think this is the buy account, not the program address of the buy asset

Reply:
programAddress is wrong

But you are right the naming is not very good too, I suggest `buyAssetReferralPubKey` and `sellAssetReferralPubKey` which is more descriptive

Changed in [6840b5c](https://github.com/shapeshift/web/pull/8267/commits/6840b5c3ff9eefd79239117bbd0c17cdb1c4bebf)

Reply:
`pubKey` is definitely much better!

Curious what's the issue with `programAddress` here? i.e program address and program public key should be interchangeable terms? 
https://coral-xyz.github.io/anchor/ts/classes/web3.PublicKey.html#findProgramAddressSync

> Find a valid program address

Reply:
I mean from a code point of view, I think `pubKey` is more descriptive of what it is: a PubKey object, if you use `address`, or `programAddress`, I would expect that it accepts a string (which is false)